### PR TITLE
CIF-1699 - [React Components] Clean-up the dependencies of the NPM module

### PR DESCRIPTION
**Pull request by dplaton**
## Description

* Convert the remaining class components to functional components
* Remove the unused plugins from the Babel configuration. Those plugins are no longer needed since we don't have class components anymore
* Remove the dependencies from `package.json`

## Related Issue

CIF-1699
## Motivation and Context

Clean up the dependency tree

## How Has This Been Tested?

Install the SNAPSHOT dependency using Venia and check that the client-side components still work (mainly the ones that use forms)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

File: react-components/package.json [hunk 1]
```diff
@@ -29,21 +29,15 @@
   "devDependencies": {
```
Comment (herzog31) [inline react-components/package.json:29]: Should we also sort the production and dev dependencies? IMO the following dependencies should be production or peer dependencies:
* `@apollo/client`
* `braintree-web-drop-in`
* `graphql`
* `i18next`
* `informed`
* `prop-types` ?
* `react`
* `react-dom`
* `react-feather`
* `react-i18next`

Comment (dplaton) [reply react-components/package.json:29]: You're right, I'll take care of it.

Comment (herzog31) [reply react-components/package.json:29]: Would still be nice to separate the dependencies accordingly. So we know which dependencies are actually needed during "runtime" and which ones are needed during "build time" only.

I think `peerDependencies` are a bonus for when we want to define the dependencies in Venia.

Comment (dplaton) [reply react-components/package.json:29]: > I think peerDependencies are a bonus for when we want to define the dependencies in Venia.

They are used to signal the consumers (not only Venia) that they absolutely need those dependencies to build their app. Also, Webpack doesn't bundle `peerDependencies`, we won't be enforcing our version of (let's say) React on consumers.

**Review by herzog31 — APPROVED** [review]

**Pull request merged**